packages/plugin-ext/src/main/browser/notebooks/notebooks-main.ts

```diff
@@ -0,0 +1,125 @@
+            disposable.dispose();
+        }
+    }
+
+    $registerNotebookSerializer(handle: number, extension: NotebookExtensionDescription, viewType: string, options: TransientOptions): void {
+        const disposables = new DisposableCollection();
+
+        disposables.push(this.notebookService.registerNotebookSerializer(viewType, extension, {
+            options,
+            dataToNotebook: async (data: BinaryBuffer): Promise<NotebookData> => {
+                const dto = await this.proxy.$dataToNotebook(handle, data, CancellationToken.None);
+                const result = NotebookDto.fromNotebookDataDto(dto);
+                return result;
+            },
+            notebookToData: (data: NotebookData): Promise<BinaryBuffer> => {
+                const result = this.proxy.$notebookToData(handle, NotebookDto.toNotebookDataDto(data), CancellationToken.None);
+                return result;
+            }
+        }));
+
+        this.notebookSerializer.set(handle, disposables);
+    }
+
+    $unregisterNotebookSerializer(handle: number): void {
+        this.notebookSerializer.get(handle)?.dispose();
+        this.notebookSerializer.delete(handle);
+    }
+
+    $emitCellStatusBarEvent(eventHandle: number): void {
+        const emitter = this.notebookCellStatusBarRegistrations.get(eventHandle);
+        if (emitter instanceof Emitter) {
+            emitter.fire(undefined);
+        }
+    }
+
+    async $registerNotebookCellStatusBarItemProvider(handle: number, eventHandle: number | undefined, viewType: string): Promise<void> {
+        const that = this;
+        const provider: NotebookCellStatusBarItemProvider = {
+            async provideCellStatusBarItems(uri: UriComponents, index: number, token: CancellationToken): Promise<NotebookCellStatusBarItemList | undefined> {
+                const result = await that.proxy.$provideNotebookCellStatusBarItems(handle, uri, index, token);
+                return {
+                    items: result?.items ?? [],
+                    dispose(): void {
+                        if (result) {
+                            that.proxy.$releaseNotebookCellStatusBarItems(result.cacheId);
+                        }
+                    }
+                };
+            },
+            viewType
+        };
+
+        if (typeof eventHandle === 'number') {
+            const emitter = new Emitter<void>();
+            this.notebookCellStatusBarRegistrations.set(eventHandle, emitter);
+            provider.onDidChangeStatusBarItems = emitter.event;
+        }
+
+        // const disposable = this._cellStatusBarService.registerCellStatusBarItemProvider(provider);
+        // this.notebookCellStatusBarRegistrations.set(handle, disposable);
+    }
+
+    async $unregisterNotebookCellStatusBarItemProvider(handle: number, eventHandle: number | undefined): Promise<void> {
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        const unregisterThing = (handle: number) => {
```

Comment:
We shouldn't shadow the variable here.

```suggestion
        const unregisterThing = (statusBarHandle: number) => {
```